### PR TITLE
Add the dist_identity primitive

### PR DESCRIPTION
**Pull request by NanmiaoWu**
Consider three cases: row-tiling, column-tiling, and sym-tiling. All of the three cases are able to deal with any number of tiles, which means they are general.

### CODE REVIEW

File: src/plugins/dist_matrixops/dist_identity.cpp [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            blaze::band(m, row_start) = T(1);
+        }
+        else if (tiling_type == "column")
+        {
+            blaze::band(m, -column_start) = T(1);
+        }
+        else if (tiling_type == "sym")
+        {
+            std::int64_t num_band = row_start - column_start;
+            std::int64_t upper_band = static_cast<std::size_t>(column_size - 1);
```
Comment (hkaiser) [inline src/plugins/dist_matrixops/dist_identity.cpp:146]: Why do you cast to a `std::size_t` before assigning to a `std::int64_t`?

Comment (NanmiaoWu) [reply src/plugins/dist_matrixops/dist_identity.cpp:146]: In line 149 and line 150, when I use max and min, the two operands in max or min should be the same type. So I cast them to std::int64_t.

Comment (NanmiaoWu) [reply src/plugins/dist_matrixops/dist_identity.cpp:146]: ah. I misunderstand your question. Seems like I dont need to cast to a std::size_t before assigning to a std::int64_t.

File: src/plugins/dist_matrixops/dist_identity.cpp [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace dist_matrixops { namespace primitives {
+    ///////////////////////////////////////////////////////////////////////////
+    execution_tree::match_pattern_type const dist_identity::match_data = {
+
+        hpx::util::make_tuple("identity_d", std::vector<std::string>{R"(
+                identity_d(
+                    _1_sz,
+                    __arg(_2_tile_index, nil),
+                    __arg(_3_numtiles, nil),
```
Comment (hkaiser) [inline src/plugins/dist_matrixops/dist_identity.cpp:46]: As @taless474 has suggested in #1147:
```suggestion
                    __arg(_2_tile_index, find_here()),
                    __arg(_3_numtiles, num_localities()),
```

Comment (NanmiaoWu) [reply src/plugins/dist_matrixops/dist_identity.cpp:46]: Will do.

**Pull request merged**